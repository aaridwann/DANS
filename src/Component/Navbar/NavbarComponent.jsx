import React from "react";

function NavbarComponent() {
  const _renderTitleCompany = () => <h2>DANS Job Portals</h2>;

  const _renderMenu = () => (
    <ul className=" flex justify-center items-center gap-4">
      <li>Login</li>
      <li>Profile</li>
      <li>Notif</li>
    </ul>
  );
  
  return (
    <nav className="mt-4 bg max-w-5xl p-4 mx-auto h-10 flex rounded-xl items-center justify-between bg-white/25 text-white">
      <_renderTitleCompany />
      <_renderMenu />
    </nav>
  );
}

export default NavbarComponent;
