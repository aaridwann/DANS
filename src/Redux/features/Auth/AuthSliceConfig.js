export const initialValue = {
  isLogin: false,
};
