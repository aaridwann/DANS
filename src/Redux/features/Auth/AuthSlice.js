import { createSlice } from "@reduxjs/toolkit";
import { initialValue } from "./AuthSliceConfig";

export const AuthSlice = createSlice({
  name: "Auth",
  initialState: initialValue,
  reducers: {
    LOGIN_ACTION: () => ({ isLogin: true }),
    LOGOUT_ACTION: () => ({ isLogin: false }),
  },
});

export const { LOGIN_ACTION, LOGOUT_ACTION } = AuthSlice.actions;
export default AuthSlice.reducer;
