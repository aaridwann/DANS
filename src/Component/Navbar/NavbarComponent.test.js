import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { useDispatch, useSelector } from "react-redux";
import { LOGOUT_ACTION } from "../../Redux/features/Auth/AuthSlice";
import NavbarComponent from "./NavbarComponent";
import "@testing-library/jest-dom/extend-expect";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: jest.fn(),
}));

describe("NavbarComponent", () => {
  const dispatch = jest.fn();
  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("should render company title and logout button when logged in", () => {
    useSelector.mockReturnValueOnce({
      auth: { isLogin: true },
    });

    render(<NavbarComponent />);

    const title = screen.getByText("DANS Job Portals");
    expect(title).toBeInTheDocument();

    const logoutButton = screen.getByText("Logout");
    expect(logoutButton).toBeInTheDocument();
  });

  test("should render company title and not render logout button when not logged in", () => {
    useSelector.mockReturnValueOnce({
      auth: { isLogin: false },
    });

    render(<NavbarComponent />);

    const title = screen.getByText("DANS Job Portals");
    expect(title).toBeInTheDocument();

    const logoutButton = screen.queryByText("Logout");
    expect(logoutButton).not.toBeInTheDocument();
  });

  test("should dispatch LOGOUT_ACTION when logout button is clicked", () => {
    useSelector.mockReturnValueOnce({
      auth: { isLogin: true },
    });

    render(<NavbarComponent />);

    const logoutButton = screen.getByText("Logout");
    fireEvent.click(logoutButton);

    expect(dispatch).toHaveBeenCalledWith(LOGOUT_ACTION());
  });
});

// Snapshot test
describe("NavbarComponent snapshot", () => {
  test("matches snapshot", () => {
    useSelector.mockReturnValue({
      auth: { isLogin: false },
    });

    const { asFragment } = render(<NavbarComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});