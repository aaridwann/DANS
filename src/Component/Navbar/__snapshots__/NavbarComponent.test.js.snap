// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent snapshot matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="mt-4 bg max-w-5xl p-4 mx-auto h-10 flex rounded-xl items-center justify-between bg-white/50 text-gray-500"
  >
    <h2>
      DANS Job Portals
    </h2>
    <ul
      class=" flex justify-center items-center gap-4"
    />
  </nav>
</DocumentFragment>
`;
